infoTree/createRecycleTree.js

```diff
@@ -0,0 +1,208 @@
+};
+Tree.prototype.findBFS = function(data) {
+  var queue = [this.root];
+  while(queue.length) {
+    var node = queue.shift();
+    if(node.data === data) {
+      return node;
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  return null;
+};
+Tree.prototype._preOrder = function(node, fn) {
+  if(node) {
+    if(fn) {
+      fn(node);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      this._preOrder(node.children[i], fn);
+    }
+  }
+};
+Tree.prototype._postOrder = function(node, fn) {
+  if(node) {
+    for(var i = 0; i < node.children.length; i++) {
+      this._postOrder(node.children[i], fn);
+    }
+    if(fn) {
+      fn(node);
+    }
+  }
+};
+Tree.prototype.traverseDFS = function(fn, method) {
+  var current = this.root;
+  if(method) {
+    this['_' + method](current, fn);
+  } else {
+    this._preOrder(current, fn);
+  }
+};
+Tree.prototype.traverseBFS = function(fn) {
+  var queue = [this.root];
+  while(queue.length) {
+    var node = queue.shift();
+    if(fn) {
+      fn(node);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+};
+Tree.prototype.print = function() {
+  if(!this.root) {
+    return console.log('No root node found');
+  }
+  var newline = new Node('|');
+  var queue = [this.root, newline];
+  var string = '';
+  while(queue.length) {
+    var node = queue.shift();
+    string += node.data.toString() + ' ';
+    if(node === newline && queue.length) {
+      queue.push(newline);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  console.log(string.slice(0, -2).trim());
+};
+Tree.prototype.printByLevel = function() {
+  if(!this.root) {
+    return console.log('No root node found');
+  }
+  var newline = new Node('\n');
+  var queue = [this.root, newline];
+  var string = '';
+  while(queue.length) {
+    var node = queue.shift();
+    string += node.data.toString() + (node.data !== '\n' ? ' ' : '');
+    if(node === newline && queue.length) {
+      queue.push(newline);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  console.log(string.trim());
+};
+//TODO: takes in a leaf data string and returns it's address as a string formatted [num, num, ... num]
+//where tree.root.children[num].children[num]...children[num].data is the string put in
+Tree.prototype.findAddress = function(data){
+	if(!this.contains(data, this.traverseBF)){
+		return console.log("Tree does not contain "+ data);
+	}
+	return true
+}
+
+Tree.prototype.findEveryLeaf = function(){
+	var leafArray = []
+	//TODO: find every leaf Node and push it into leafArray
+	return leafArray;
+}
+
+//finds the URL to scrape data from by adding parent's name to base URL
+Tree.prototype.findUrl = function(leaf){
+	url = "https://www.seattle.gov/util/MyServices/WhereDoesItGo/"
```

Comment:
no variable declaration for `url`, this makes it a global which we definitely want to avoid.